internal/inventory/inventory.go

```diff
@@ -0,0 +1,126 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package inventory
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	libevents "github.com/elastic/beats/v7/libbeat/beat/events"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/samber/lo"
+)
+
+type AssetInventory struct {
+	fetchers            []AssetFetcher
+	publisher           AssetPublisher
+	bufferFlushInterval time.Duration
+	bufferMaxSize       int
+	logger              *logp.Logger
+	assetCh             chan AssetEvent
+	now                 func() time.Time
+}
+
+type AssetFetcher interface {
+	Fetch(ctx context.Context, assetChannel chan<- AssetEvent)
+}
+
+type AssetPublisher interface {
+	PublishAll([]beat.Event)
+}
+
+func NewAssetInventory(logger *logp.Logger, fetchers []AssetFetcher, publisher AssetPublisher, now func() time.Time) AssetInventory {
+	logger.Info("Initializing Asset Inventory POC")
+	return AssetInventory{
+		logger:    logger,
+		fetchers:  fetchers,
+		publisher: publisher,
+		// move to a configuration parameter
```

Comment:
This looks like a leftover TODO

Reply:
Jah, it's indeed. For the poc I decided to keep it here, I don't see a reason to make it an external configuration.

Do you disagree?

Reply:
Makes sense to me. Just making sure you haven't missed it by accident :)